gadget/layout.go

```diff
@@ -189,6 +192,69 @@ func LayoutVolumePartially(volume *Volume, constraints LayoutConstraints) (*Part
 	return vol, nil
 }
 
+func resolveOne(gadgetRootDir, kernelRootDir string, kernelInfo *KernelInfo, pathOrRef string) (string, error) {
+	// content may refer to "$kernel:<name>/<content>"
+	if strings.HasPrefix(pathOrRef, "$kernel:") {
+		kernelRef := strings.SplitN(pathOrRef, ":", 2)[1]
```

Comment:
We already know that prefix exists, so we could have`kernelRef := pathOrRef[len("$kernel:"):]`